frontend/src/pages/org/browser-profiles-list.ts

```diff
@@ -124,84 +136,84 @@ export class BrowserProfilesList extends BtrixElement {
     },
   );
 
-  private readonly filterByTags = new SearchParamsValue<string[] | undefined>(
-    this,
-    (value, params) => {
-      params.delete("tags");
-      value?.forEach((v) => {
-        params.append("tags", v);
-      });
-      return params;
-    },
-    (params) => params.getAll("tags"),
-  );
-
-  private readonly filterByTagsType = new SearchParamsValue<"and" | "or">(
+  private readonly filterBy = new SearchParamsValue<FilterBy>(
     this,
     (value, params) => {
-      if (value === "and") {
-        params.set("tagsType", value);
+      if ("name" in value && value["name"]) {
+        params.set("name", value["name"]);
+      } else {
+        params.delete("name");
+      }
+      if ("tags" in value) {
+        params.delete("tags");
+        value["tags"]?.forEach((v) => {
+          params.append("tags", v);
+        });
+      } else {
+        params.delete("tags");
+      }
+      if ("tagsType" in value && value["tagsType"] === "and") {
+        params.set("tagsType", value["tagsType"]);
       } else {
         params.delete("tagsType");
       }
-      return params;
-    },
-    (params) => (params.get("tagsType") === "and" ? "and" : "or"),
-  );
-
-  private readonly filterByCurrentUser = new SearchParamsValue<boolean>(
-    this,
-    (value, params) => {
-      if (value) {
+      if ("mine" in value && value["mine"]) {
         params.set("mine", "true");
+        window.sessionStorage.setItem(
+          FILTER_BY_CURRENT_USER_STORAGE_KEY,
+          "true",
+        );
       } else {
         params.delete("mine");
+        window.sessionStorage.removeItem(FILTER_BY_CURRENT_USER_STORAGE_KEY);
       }
       return params;
     },
-    (params) => params.get("mine") === "true",
+    (params) => ({
+      name: params.get("name") || undefined,
+      tags: params.getAll("tags"),
+      tagsType: params.get("tagsType") === "and" ? "and" : "or",
+      mine: params.get("mine") === "true",
+    }),
     {
-      initial: (initialValue) =>
-        window.sessionStorage.getItem(FILTER_BY_CURRENT_USER_STORAGE_KEY) ===
-          "true" ||
-        initialValue ||
-        false,
+      initial: (initialValue) => ({
+        ...initialValue,
+        mine:
+          window.sessionStorage.getItem(FILTER_BY_CURRENT_USER_STORAGE_KEY) ===
+            "true" ||
+          initialValue?.["mine"] ||
+          false,
+      }),
     },
   );
 
   private get hasFiltersSet() {
-    return [
-      this.filterByCurrentUser.value || undefined,
-      this.filterByTags.value?.length || undefined,
-    ].some((v) => v !== undefined);
+    const filterBy = this.filterBy.value;
+    return (
+      filterBy.name ||
+      filterBy.tags?.length ||
+      filterBy.tagsType !== DEFAULT_TAGS_TYPE ||
```

Comment:
When resetting `filterBy` to an empty object, `filterBy.tagsType !== DEFAULT_TAGS_TYPE` evaluates to `true`, so the Clear All button doesn't disappear when clicked. Here's one approach, it could also be done with a check for undefined before comparing against `DEFAULT_TAGS_TYPE`.
```suggestion
      (filterBy.tagsType ?? DEFAULT_TAGS_TYPE) !== DEFAULT_TAGS_TYPE ||
```

Reply:
Nice catch!